test/suites/dev/test-xcm-v2/test-mock-hrmp-transact-ethereum-12.ts

```diff
@@ -0,0 +1,158 @@
+
+        for (const xcmTransaction of xcmTransactions) {
+          expectedTransferredAmount += amountToTransfer;
+          expectedTransferredAmountPlusFees += amountToTransfer + targetXcmFee;
+          // TODO need to update lookup types for xcm ethereum transaction V2
+          const transferCall = context.polkadotJs().tx.ethereumXcm.transact(xcmTransaction as any);
+          const transferCallEncoded = transferCall?.method.toHex();
+
+          // We are going to test that we can receive a transact operation from parachain 1
+          // using descendOrigin first
+          const xcmMessage = new XcmFragment({
+            assets: [
+              {
+                multilocation: {
+                  parents: 0,
+                  interior: {
+                    X1: { PalletInstance: balancesPalletIndex },
+                  },
+                },
+                fungible: targetXcmFee,
+              },
+            ],
+            weight_limit: {
+              refTime: targetXcmWeight,
+              proofSize: (GAS_LIMIT / GAS_LIMIT_POV_RATIO) * 2,
+            } as any,
+            descend_origin: sendingAddress,
+          })
+            .descend_origin()
+            .withdraw_asset()
+            .buy_execution()
+            .push_any({
+              Transact: {
+                originKind: "SovereignAccount",
+                // 500_000 gas limit + db read
+                requireWeightAtMost: {
+                  refTime: 12_525_000_000,
+                  proofSize: GAS_LIMIT / GAS_LIMIT_POV_RATIO,
+                },
+                call: {
+                  encoded: transferCallEncoded,
+                },
+              },
+            })
+            .as_v3();
+
+          // Send an XCM and create block to execute it
+          await injectHrmpMessageAndSeal(context, 1, {
+            type: "XcmVersionedXcm",
+            payload: xcmMessage,
+          } as RawXcmMessage);
+
+          // Make sure the state has ALITH's foreign parachain tokens
+          const testAccountBalance = (
+            await context.polkadotJs().query.system.account(random.address)
+          ).data.free.toBigInt();
+          expect(testAccountBalance).to.eq(expectedTransferredAmount);
+
+          // Make sure ALITH has been deducted fees once (in xcm-executor) and balance has been
+          // transfered through evm.
+          const alithAccountBalance = await context.viem().getBalance({ address: descendAddress });
+          expect(BigInt(alithAccountBalance)).to.eq(
+            transferredBalance - expectedTransferredAmountPlusFees
+          );
+
+          const weightBlock = await context.polkadotJs().query.system.blockWeight();
+          // Make sure the system block weight corresponds to gas used and not gas limit
+          // It should be sufficient to verify that we used less than what was marked
+          expect(12_500_000_000n + 25_000_000n - weightBlock.mandatory.refTime.toBigInt() > 0n).to
```

Comment:
this can instead be:
```
 expect(12_500_000_000n + 25_000_000n - weightBlock.mandatory.refTime.toBigInt()).toBeGreaterThan(0n)
```
Which will give a much nicer error if it breaks